hypothesis-python/src/hypothesis/internal/conjecture/choice.py

```diff
@@ -123,7 +180,7 @@ def zigzag_value(index, *, shrink_towards):
     return shrink_towards + n
 
 
-def choice_to_index(choice, kwargs):
+def choice_to_index(choice: ChoiceT, kwargs: ChoiceKwargsT) -> int:
```

Comment:
We could avoid casts by instead using overloads here; that would in turn place more precise constraints on callers.  (I don't think this is worth doing, but we could leave a note to come back and check again later)

Reply:
actually I'd be very happy to do this, the (albeit negligible) performance hit from casting makes me sad. Will come back in a later pull.